src/main/java/mekanism/common/CommonPlayerTickHandler.java

```diff
@@ -124,23 +119,24 @@ private void tickEnd(Player player) {
             }
         }
 
-        ItemStack chest = player.getItemBySlot(EquipmentSlot.CHEST);
-        ItemStack jetpackStack = ItemJetpack.getJetpack(player, chest);
-        JetpackMode mode = getJetpackModeIfOn(player, jetpackStack);
-        if (mode != JetpackMode.DISABLED) {
-            if (handleJetpackMotion(player, mode, () -> Mekanism.keyMap.has(player.getUUID(), KeySync.ASCEND))) {
-                player.fallDistance = 0.0F;
-                if (player instanceof ServerPlayer serverPlayer) {
-                    serverPlayer.connection.aboveGroundTickCount = 0;
+        ItemStack jetpack = IJetpackItem.getActiveJetpack(player);
+        if (!jetpack.isEmpty()) {
+            ItemStack primaryJetpack = IJetpackItem.getPrimaryJetpack(player);
+            JetpackMode primaryMode = ((IJetpackItem) primaryJetpack.getItem()).getJetpackMode(primaryJetpack);
+            JetpackMode mode = IJetpackItem.getPlayerJetpackMode(player, primaryMode);
+            if (mode != JetpackMode.DISABLED) {
+                // Client handles motion and sets fallDistance in ClientTickHandler at the tick start
```

Comment:
This is just condensed return logic for the handle motion call, so if we need to change this back it should just be replacing the method out. I almost left it just so it wouldn't be a hassle to find if the logic changed for some reason.

Speaking of though, I think I finally figured out why vanilla doesn't let you stop using an elytra mid flight (bedrock does and running into something also does it automatically). I was curious how much fall damage or energy would be used falling 1k+ meters in the mekasuit and it turns out not much since it disconnects you every 200 meters (resetting your fall damage) as flying is not enabled on the server.

Reply:
I think I would definitely prefer you calling the method on the server as well as I recall with some things related to boosting the server having issues at times due to the moved wrongly/too quickly thing and this way the server would know they are supposed to be moving quicker.

I am not quite sure what you are talking about the elytra and being kicked for flying though?

Reply:
I left the comment as I think it still explains why we are only doing this for the server.

The flying/kick thing isn't really something we can fix so I guess not especially relevant, was more just stating that as you can stop elytra flight with the mekasuit it is easy to fall 200m and disconnect. But I guess that problem exists with just the jetpack too.